### PR TITLE
Fix attachment right click save as

### PR DESCRIPTION
**Pull request by Bilb**
Fixes #2893 #2890

### CODE REVIEW

File: ts/components/conversation/message/message-content/MessageContextMenu.tsx [hunk 1]
```diff
@@ -313,6 +278,29 @@ export const MessageContextMenu = (props: Props) => {
     }
   };
 
+  const saveAttachment = (e: ItemParams) => {
+    // this is quite dirty but considering that we want the context menu of the message to show on click on the attachment
+    // and the context menu save attachment item to save the right attachment I did not find a better way for now.
+    const targetAttachmentIndex = isNumber(e?.props?.dataAttachmentIndex)
+      ? e.props.dataAttachmentIndex
+      : 0;
+    e.event.stopPropagation();
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageContextMenu.tsx:288]: Should this not be on the first line of the function?

File: ts/components/conversation/message/message-content/MessageContextMenu.tsx [hunk 2]
```diff
@@ -313,6 +278,29 @@ export const MessageContextMenu = (props: Props) => {
...
+    if (!attachments?.length || !convoId || !sender) {
+      return;
+    }
+
+    if (targetAttachmentIndex > attachments.length) {
+      return;
+    }
+    const messageTimestamp = timestamp || serverTimestamp || 0;
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageContextMenu.tsx:296]: new line above

File: ts/components/conversation/message/message-content/MessageContextMenu.tsx [hunk 3]
```diff
@@ -360,18 +348,14 @@ export const MessageContextMenu = (props: Props) => {
...
-        >
+        <Menu id={contextMenuId} onVisibilityChange={onVisibilityChange} animation="fade">
           {enableReactions && (
             // eslint-disable-next-line @typescript-eslint/no-misused-promises
             <MessageReactBar action={onEmojiClick} additionalAction={onShowEmoji} />
           )}
-          <SaveAttachment messageId={messageId} />
-
+          {attachments?.length ? (
+            <Item onClick={saveAttachment}>{window.i18n('downloadAttachment')}</Item>
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageContextMenu.tsx:358]: Why not `&&` here instead? Then you don't need to return `null`?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/components/conversation/message/message-content/MessageContextMenu.tsx (hunk 3 above)
Comment (Bilb) [reply ts/components/conversation/message/message-content/MessageContextMenu.tsx:358]: that'd be the "pitfall" of https://react.dev/learn/conditional-rendering#conditionally-including-jsx

**Pull request merged**